adds active restriction lisit to visitor dashboard

lspdrz opened the pull request:
DO NOT MERGE UNTIL https://github.com/Ameelio/connect-api-nest/pull/526 is merged and deployed

[sc-1491]

![image](https://user-images.githubusercontent.com/9705113/144096331-a1ec707f-e0af-47e9-b007-afa0114e4ea2.png)

gabrielsaruhashi commented on src/components/Profile/ProfileHeader/ContactProfileHeader.tsx:
```diff
@@ -1,16 +1,25 @@
+  const { loading, error, data } = useQuery(GetVisitorWithRestrictionsDocument, {
+    variables: { visitorId: contact.id },
+  });
+  console.log('hi data')
```
✂️

lspdrz replied:
Done!!

gabrielsaruhashi commented on src/components/Profile/ProfileHeader/ContactProfileHeader.tsx:
```diff
@@ -1,16 +1,24 @@
 }
 
 const ContactProfileHeader = ({ contact, facilityName }: Props) => {
+  const { loading, error, data } = useQuery(GetVisitorWithRestrictionsDocument, {
```
we should keep components from src dummy. 
Let's pass it down as props and do this query at the top-level ContactPage.
It's probably an opportunity to start using the Visitor graphql type here instead of the old Contact (can be a fast-follow PR)
The Visitor graphql type has all the info Contact has

lspdrz replied:
Done with the above. The Visitor graphql type doesn't have all the info Contact has, such as dateOfBirth. Down to take care in another story

gabrielsaruhashi reviewed (changes requested):
some comments!

gabrielsaruhashi commented on src/graphql/GetVisitorWithRestrictions.graphql:
```diff
@@ -0,0 +1,17 @@
+        ...InmateFragment
+      }
+      restrictions {
+        id
```
RestrictionFragment?

gabrielsaruhashi commented on src/pages/ContactPage.tsx:
```diff
@@ -40,7 +45,7 @@ function ContactPage({ match }: RouteComponentProps<TParams>): ReactElement {
     <Layout>
       <Layout.Content style={WRAPPER_STYLE}>
-        <ContactProfileHeader contact={contact} facilityName={facilityName} />
+        <ContactProfileHeader contact={contact} connections={data?.visitor.connections} facilityName={facilityName} />
```
data?.visitor.connections || []

no need to pass down an undefined

gabrielsaruhashi commented on src/components/Profile/ProfileHeader/ContactProfileHeader.tsx:
```diff
@@ -1,16 +1,23 @@
+type Connection = GetVisitorWithRestrictionsQuery['visitor']['connections'][0]
 interface Props {
   contact: Contact;
+  connections: Connection[] | undefined;
```
connections: Connection[]

you should have a fallback to prevent undefined from being passed

gabrielsaruhashi commented on src/components/Profile/ProfileHeader/ContactProfileHeader.tsx:
```diff
@@ -60,6 +80,22 @@ const ContactProfileHeader = ({ contact, facilityName }: Props) => {
+        {facilityhasMessaging && <Row className="flex space-x-4 flex-wrap">
+          <Typography.Text>Disabled Privileges: </Typography.Text>
+          {
+            connections && connections.map(
```
this wont be needed once you have the fallback

gabrielsaruhashi commented on src/components/Profile/ProfileHeader/ContactProfileHeader.tsx:
```diff
@@ -60,6 +80,22 @@ const ContactProfileHeader = ({ contact, facilityName }: Props) => {
+            connections && connections.map(
+              connection => connection.restrictions.map(
+                restriction => 
+                <div>
```
you're missing a key

gabrielsaruhashi commented on src/components/Profile/ProfileHeader/ContactProfileHeader.tsx:
```diff
@@ -60,6 +80,22 @@ const ContactProfileHeader = ({ contact, facilityName }: Props) => {
+          {
+            connections && connections.map(
+              connection => connection.restrictions.map(
+                restriction => 
```
is prettier not being run upon committing?
this doesn't look formatted

lspdrz replied:
should be? I wrapped in parentheses hopefully it'll help it

gabrielsaruhashi commented on src/components/Profile/ProfileHeader/ContactProfileHeader.tsx:
```diff
@@ -27,6 +34,19 @@ const ContactProfileHeader = ({ contact, facilityName }: Props) => {
+        return 'Calls'
+      case (MeetingType.InPersonContact):
+        return 'In-Person Visits, with Contact'
+      case (MeetingType.InPersonNoContact):
```
so currently we cannot create a connection restriction for e-messaging?

gabrielsaruhashi replied:
it'd be an important use case.
that's the one that's driven us to create this in the first place

lspdrz replied:
We don't have the Meeting Type for emessaging in the backend yet

gabrielsaruhashi commented on src/graphql/GetVisitorWithRestrictions.graphql:
```diff
@@ -0,0 +1,17 @@
+query GetVisitorWithRestrictions($visitorId: ID!) {
+  visitor(id: $visitorId) {
+    id
+    firstName
```
you can use VisitorFragment here

lspdrz replied:
Can we? I want to include the restrictions and inmate foreign keys here but they're not on the fragment. We probably don't want to load those objects all the time either so I don't want to move to the fragment

gabrielsaruhashi reviewed (approved):
could you please address the comments? and or make sure we address them on the next PR

lspdrz commented:
> could you please address the comments? and or make sure we address them on the next PR

Implemented requested changes and addressed a couple of comments. Will merge when tests pass.

Regarding the emessaging type: I'm going to add the meeting type to backend and uncomment the frontend features in one story. We need to think about the repercussions of adding another meeting type enum value will have to other features on the project
